lib/ship_quiet_logistics/handlers/error_message_handler.rb

```diff
@@ -4,7 +4,8 @@ class ErrorMessageHandler
       def self.call(message)
         shipment = Spree::Shipment.find_by(number: message.shipment_number)
 
-        shipment.update(pushed: false)
+        shipment.update(pushed: false) if shipment
```

Comment:
Should this fail silently if the shipment can't be found?

Reply:
The issue I got is that the error_message_handle is generic for every type of request:
https://github.com/DynamoMTL/ship_quiet_logistics/blob/master/lib/ship_quiet_logistics.rb#L45

Thus, in the case of an inventory summary request, there is no shipment to look up and it broke because of that. A better design I think would be to have a error handler per type of request, but I feel it out of scope for this PR.

Reply:
I don't think it's necessarily out of scope. The scope of this PR is handling inventory-summary documents, part of which is handling errors in those.

Reply:
What do you think of this proposal to manage different type of errors : https://github.com/DynamoMTL/ship_quiet_logistics/pull/10/commits/1af9f510e5b0dfb62613e95c8eff6e2d520f6776 ?